app/assets/javascripts/spreadsheet.js

```diff
@@ -1,71 +1,97 @@
-	var input_controls = [$("#filetypes_1"), $("#filetypes_2"), $("#convert_only"), $("#note_text")]
-	var radio_buttons = input_controls.slice(0, 3)
+    $.fn.argoSpreadsheet = function () {
+        var $el;             // Use dollar sign for JQuery objects
+        var $submitButton;
 	
-	// Adjust the width of the lightbox when it hosts a metadata spreadsheet upload form.
-	$("#spreadsheet-upload-container").parent().parent().css("width", "650px");
-
-	// Width of the lightbox when hosting a metadata bulk upload log
-	$("#spreadsheet-log-container").parent().parent().css("width", "700")
+	var enableSubmit = function() {
+	    $submitButton.prop('disabled', false);
+	};
 	
-	// None of the form controls should be functional until a file has been selected
-	$("#spreadsheet_submit").prop("disabled", true);
-	input_controls.map(disable_control)
-
-	// Enable everything except for the submit button upon file upload
-	$("#spreadsheet_file").change(function (){
-	    input_controls.map(enable_control)
-	});
+	var disableControl = function(ctrl) {
+	    ctrl.prop('disabled', true);
+	};
+	
+	var enableControl = function(ctrl) {
+	    ctrl.prop('disabled', false);
+	};
 
-	// Only when the user has uploaded a file AND selected one of the radio buttons should it be possible to submit
-	radio_buttons.forEach(function(button) {
-	    button.click(function() {
-		enable_submit()
+	return this.each(function() {
+	    $el = $(this);     // $el is now spreadsheet-upload-container
+	    $submitButton = $el.find('#spreadsheet_submit');
+	    
+	    // When the user clicks the 'MODS bulk loads' button, a lightbox is
+	    // opened. The event 'loaded.blacklight.ajax-modal' is fired just
+	    // before this Blacklight lightbox is shown.
+	    var inputControls = [$('#filetypes_1'),
+				 $('#filetypes_2'),
+				 $('#convert_only'),
+				 $('#note_text')];
+	    var radioButtons = inputControls.slice(0, 3);
+	    
+	    // Adjust the width of the lightbox when it hosts a metadata
+	    // spreadsheet upload form.
+	    $el.parent().parent().css('width', '650px');
+	    
+	    // Width of the lightbox when hosting a metadata bulk upload log
+	    $el.parent().parent().css('width', '700');
+	    
+	    // None of the form controls should be functional until a file has been
+	    // selected
+	    $submitButton.prop('disabled', true);
+	    inputControls.map(disableControl);
+	    
+	    // Enable everything except for the submit button upon file upload
+	    $el.find('#spreadsheet_file').change(function (){
+		inputControls.map(enableControl);
+	    });
+	    
+	    // Only when the user has uploaded a file AND selected one of the
+	    // radio buttons should it be possible to submit
+	    radioButtons.forEach(function(button) {
+		button.click(function() {
+		    enableSubmit();
+		});
 	    });
 	});
+    };
+})(jQuery);	
+	
+
+// The Blacklight onLoad event works better than the regular onLoad event if
+// turbolinks is enabled.
+Blacklight.onLoad(function(){
```

Comment:
This can be fixed by doing something like this at top of file:

https://github.com/sul-dlss/argo/blob/develop/app/assets/javascripts/modules/index_queue.js#L1